controllers/buildpipeline/buildpipeline_adapter.go

```diff
@@ -277,20 +281,45 @@ func (a *Adapter) getSucceededBuildPipelineRunsForComponent(component *applicati
 	return &succeededPipelineRuns, nil
 }
 
-func (a *Adapter) annotateBuildPipelineRunWithSnapshot(pipelineRun *tektonv1.PipelineRun, snapshot *applicationapiv1alpha1.Snapshot) (*tektonv1.PipelineRun, error) {
+func (a *Adapter) annotateBuildPipelineRun(pipelineRun *tektonv1.PipelineRun, key, value string) (*tektonv1.PipelineRun, error) {
 	patch := client.MergeFrom(pipelineRun.DeepCopy())
 
-	_ = metadata.SetAnnotation(&pipelineRun.ObjectMeta, tekton.SnapshotNameLabel, snapshot.Name)
+	_ = metadata.SetAnnotation(&pipelineRun.ObjectMeta, key, value)
 
 	err := a.client.Patch(a.context, pipelineRun, patch)
 	if err != nil {
 		return pipelineRun, err
 	}
-	a.logger.LogAuditEvent("Updated build pipelineRun", pipelineRun, h.LogActionUpdate,
-		"snapshot.Name", snapshot.Name)
 	return pipelineRun, nil
 }
 
+func (a *Adapter) annotateBuildPipelineRunWithSnapshot(pipelineRun *tektonv1.PipelineRun, snapshot *applicationapiv1alpha1.Snapshot) (*tektonv1.PipelineRun, error) {
+	pipelineRun, err := a.annotateBuildPipelineRun(pipelineRun, tekton.SnapshotNameLabel, snapshot.Name)
+	if err != nil {
+		a.logger.LogAuditEvent("Updated build pipelineRun", pipelineRun, h.LogActionUpdate,
+			"snapshot.Name", snapshot.Name)
+	}
+	return pipelineRun, err
+}
+
+func (a *Adapter) annotateBuildPipelineRunWithCreateSnapshotAnnotation(status string, err error) {
```

Comment:
Why  we don't return error state here? We can at least log it in defer func when something fails

Reply:
I didn't return an error because I thought it would be thrown away in defer.  Logging is a good point, I'll update that.